templates/app-link.tpl.php

```diff
@@ -31,10 +31,23 @@
 <?php endforeach; ?>
 
 <script>
+var REFERRER = document.referrer;
 var PLATFORM_INFO = <?php print $platform_info; ?>;
 var PLATFORM_DATA = <?php print $platform_data; ?>;
 var FALLBACK_URL = <?php print $fallback_url; ?>;
 
+// Check and see if 'referrer is already a part of the fallback URL.
+// If it isn't, then check to see there is already a querystring. Then
+// add REFERRER based on that result.
+if (FALLBACK_URL.indexOf('referrer') == -1 && REFERRER.length > 0) {
+  if (FALLBACK_URL.indexOf('?') > -1) {
+    FALLBACK_URL = FALLBACK_URL + '&referrer=' + encodeURIComponent(REFERRER);
+  }
+  else {
+    FALLBACK_URL = FALLBACK_URL + '?referrer=' + encodeURIComponent(REFERRER);
+  }
+}
+
```

Comment:
We appear to be repeating this logic.  Here, and then again in the platform specific JS.  Since this once ignores the "Support QS" rule, let's just remove this version.  This will keep the code smaller too.  To help out, I'll go ahead and push up a commit.